src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -76,8 +81,10 @@ func (r *StatefulSetResource) Ensure(ctx context.Context) error {
 	r.LastObservedState = &sts
 
 	// Ensure StatefulSet #replicas equals cluster requirement.
-	if sts.Spec.Replicas != r.pandaCluster.Spec.Replicas {
+	if sts.Spec.Replicas != nil && r.pandaCluster.Spec.Replicas != nil && *sts.Spec.Replicas != *r.pandaCluster.Spec.Replicas {
+		logger.Info(fmt.Sprintf("StatefulSet %s has replicas set to %d but need %d. Going to update", r.Key().Name, *sts.Spec.Replicas, *r.pandaCluster.Spec.Replicas))
 		sts.Spec.Replicas = r.pandaCluster.Spec.Replicas
+
```

Comment:
ditto

Reply:
if there's multiple statements pre if linter requires this

Reply:
I was thinking more of and log, empty line, assignment and last if statement like:
```
		logger.Info(fmt.Sprintf("StatefulSet %s has replicas set to %d but need %d. Going to update", r.Key().Name, *sts.Spec.Replicas, *r.pandaCluster.Spec.Replicas))

sts.Spec.Replicas = r.pandaCluster.Spec.Replicas
if err := r.Update(ctx, &sts); err != nil {
```

Reply:
This new line is unrelated to PR